src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -2,20 +2,96 @@
 
 import org.junit.Before;
 import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
 
-import static org.junit.Assert.assertTrue;
+import java.math.BigDecimal;
+import java.util.NoSuchElementException;
 
-public class ArrayListProductDaoTest
-{
-    private ProductDao productDao;
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+public class ArrayListProductDaoTest {
```

Comment:
Add at least one test which demonstrates the filtering logic. That means you have to set up mocks that:
- there are 2+ products contained in the database (product list)
- there are 1+ of them having null price or 0 stock
- verify that all the others are returned by find()